### ISSUE TITLE
Using mergeProps in data causing TypeError

### ISSUE THREAD

**Issue by martinpengellyphillips** [solidjs/solid-app-router#13]
Possibly similar to #6 - using `mergeProps` in data seems to cause the following error on latest solid-js and solid-app-router:

```
TypeError
can't define property Symbol("state-node"): Object is not extensible
getDataNodes
https://uojg1.csb.app/node_modules/solid-js/dist/solid.js:1131:23
get
https://uojg1.csb.app/node_modules/solid-js/dist/solid.js:1158:33
Data/</<
tab1.tsx:12:18

   9 |       <button type="button" onClick={increment}>
  10 |         {count()}
  11 |       </button>
> 12 |       <code>{JSON.stringify(props, null, 2)}</code>
     |                  ^
  13 |     </>
  14 |   );
  15 | }
```

Here is the forked sandbox from #6 where I just updated the versions to latest `solid-js` and `solid-app-router`: https://codesandbox.io/s/silent-voice-uojg1

Removing the use of `mergeProps` allows it to work without error.

**Comment by ryansolid** [comment]
Thanks for reporting. It looks to be a case where deep wrapping of reactivity is exposing internals that should never be wrapped (and in this case happen to be frozen). JSON.stringify just triggers this. I'm debating best ways to handling this since this is not the type of thing that will be limited to this case.

**Comment by ryansolid** [comment]
There is a second issue here, after fixing the first. Router has a data prop which is the resolved data components. `JSON.stringify` is caught in an infinite loop since `mergeProps` is feeding it into itself. I've made data/current non-enumerable to fix this. Fixed in 0.44.0.

**Comment by martinpengellyphillips** [comment]
Nice one - thank you!
